WootricSDK/WootricSDK/WTRApiClient.m

```diff
@@ -90,23 +90,26 @@ - (void)getEndUserWithEmail:(void (^)(NSInteger endUserID))endUserWithID {
             NSLog(@"WootricSDK (GET end user): %@", error);
         }
         else {
-            NSArray *responseJSON = [NSJSONSerialization JSONObjectWithData:data options:0 error:nil];
+          id responseJSON = [NSJSONSerialization JSONObjectWithData:data options:0 error:nil];
+          if ([responseJSON isKindOfClass:[NSArray class]]) {
             if ([responseJSON count] == 0) {
-                [self createEndUser:^(NSInteger endUserID) {
-                    endUserWithID(endUserID);
-                }];
-            }
-            else {
-                NSDictionary *endUser = responseJSON[0];
-                
-                if (endUser[@"id"]) {
-                    NSInteger endUserID = [endUser[@"id"] integerValue];
-                    if (!_endUserAlreadyUpdated) {
-                        [self updateExistingEndUser:endUserID];
-                    }
-                    endUserWithID(endUserID);
+              [self createEndUser:^(NSInteger endUserID) {
+                endUserWithID(endUserID);
+              }];
+            } else {
+              NSDictionary *endUser = responseJSON[0];
+              
+              if (endUser[@"id"]) {
+                NSInteger endUserID = [endUser[@"id"] integerValue];
+                if (!_endUserAlreadyUpdated) {
+                  [self updateExistingEndUser:endUserID];
                 }
+                endUserWithID(endUserID);
+              }
             }
+          } else {
+            NSLog(@"WootricSDK - Error: Server response is not an array. Response object: %@. Class: %@", responseJSON, [responseJSON class]);
```

Comment:
Update this NSLog since we know that when we get a dictionary it's an error from the API.